web/src/features/manual-scoring/components/CreateScoreConfigButton.tsx

```diff
@@ -35,30 +34,62 @@ import {
   isBooleanDataType,
   isCategoricalDataType,
   isNumericDataType,
+  isPresent,
 } from "@/src/features/manual-scoring/lib/helpers";
 import DocPopup from "@/src/components/layouts/doc-popup";
 import { usePostHogClientCapture } from "@/src/features/posthog-analytics/usePostHogClientCapture";
+import { z } from "zod";
 
-const availableDataTypes = [
-  ScoreDataType.NUMERIC,
-  ScoreDataType.CATEGORICAL,
-  ScoreDataType.BOOLEAN,
-] as const;
-
-const category = z.object({
-  label: z.string().min(1),
-  value: z.coerce.number(),
-});
-
-const formSchema = z.object({
+const createConfigSchema = z.object({
   name: z.string().min(1).max(35),
   dataType: z.enum(availableDataTypes),
   minValue: z.coerce.number().optional(),
   maxValue: z.coerce.number().optional(),
-  categories: z.array(category).optional(),
+  categories: z.array(Category).optional(),
   description: z.string().optional(),
 });
 
+type CreateConfig = z.infer<typeof createConfigSchema>;
+
+const validateScoreConfig = (values: CreateConfig): string | null => {
```

Comment:
see above, does it make sense to keep this check in a central place and use for ui, trpc route and api?